pkg/gardenlet/controller/backupbucket/backup_bucket_reconciler.go

```diff
@@ -72,7 +75,7 @@ func (r *reconciler) Reconcile(ctx context.Context, request reconcile.Request) (
 	if bb.Spec.SeedName == nil {
 		message := "Cannot reconcile BackupBucket: Waiting for BackupBucket to get scheduled on a Seed"
 		r.recorder.Event(bb, corev1.EventTypeWarning, "OperationPending", message)
-		return reconcile.Result{}, utilerrors.WithSuppressed(fmt.Errorf("backupBucket %s has not yet been scheduled on a Seed", bb.Name), updateBackupBucketStatusPending(ctx, r.gardenClient, bb, message))
+		return reconcile.Result{}, utilerrors.WithSuppressed(fmt.Errorf("backupBucket %s has not yet been scheduled on a Seed", bb.Name), updateBackupBucketStatusPending(ctx, r.gardenClient, bb, message, r.clock))
 	}
```

Comment:
Can we delete this whole block? The gardenlet will only watch `BackupBucket`s whose `.spec.seedName` is equal to the seed it is responsible for. Hence, this if condition would never be reached?